test/functional/pdo_sqlsrv/pdo_errorinfo_emulateprepare.phpt

```diff
@@ -1,108 +1,87 @@
 --TEST--
 Test errorInfo when prepare with and without emulate prepare
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require('skipif_mid-refactor.inc'); ?>
 --FILE--
 <?php
-
-require_once 'MsSetup.inc';
-
-$conn = new PDO("sqlsrv:server=$server;Database=$databaseName", $uid, $pwd, array(PDO::ATTR_ERRMODE => PDO::ERRMODE_WARNING));
-
-//drop, create and insert
-$conn->query("IF OBJECT_ID('dbo.test_table', 'U') IS NOT NULL DROP TABLE dbo.test_table");
-$conn->query("CREATE TABLE [dbo].[test_table](c1 int, c2 int)");
-$conn->query("INSERT INTO [dbo].[test_table] VALUES (1, 10)");
-$conn->query("INSERT INTO [dbo].[test_table] VALUES (2, 20)");
-
-echo "\n****testing with emulate prepare****\n";
-$stmt = $conn->prepare("SELECT c2 FROM test_table WHERE c1= :int", array(PDO::ATTR_EMULATE_PREPARES => true));
-
-$int_col = 1;
-//bind param with the wrong parameter name to test for errorInfo
-$stmt->bindParam(':in', $int_col);
-
-$stmt->execute();
-
-echo "Statement error info:\n";
-print_r($stmt->errorInfo());
-
-if ($stmt->errorInfo()[1] == NULL && $stmt->errorInfo()[2] == NULL) {
-	echo "stmt native code and native message are NULL.\n";
-}
-else {
-	echo "stmt native code and native message should be NULL.\n";
+require_once("MsCommon_mid-refactor.inc");
+
+try {
+    $conn = connect();
+    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_SILENT);
+
```

Comment:
shouldn't the ERRMODE be WARNING?

Reply:
If it's WARNING, it'll print the WARNING without us asking for it and the warning is different between ae and non ae so can't compare the warning in the EXPECTIF section.